3rdparty/libcxx/CMakeLists.txt

```diff
@@ -1,85 +1,72 @@
+  libcxx/src/thread.cpp
+  libcxx/src/typeinfo.cpp
+  libcxx/src/utility.cpp
+  libcxx/src/valarray.cpp
+  libcxx/src/variant.cpp
+  libcxx/src/vector.cpp)
 
-add_library(oelibcxx_cxx STATIC
-    __dso_handle.cpp
+add_dependencies(libcxx libcxx_includes)
 
-    libcxx/src/algorithm.cpp
-    libcxx/src/any.cpp
-    libcxx/src/bind.cpp
-    libcxx/src/chrono.cpp
-    libcxx/src/condition_variable.cpp
-    libcxx/src/debug.cpp
-    libcxx/src/exception.cpp
-    libcxx/src/functional.cpp
-    libcxx/src/future.cpp
-    libcxx/src/hash.cpp
-    libcxx/src/ios.cpp
-    libcxx/src/iostream.cpp
-    libcxx/src/locale.cpp
-    libcxx/src/memory.cpp
-    libcxx/src/mutex.cpp
-    libcxx/src/new.cpp
-    libcxx/src/optional.cpp
-    libcxx/src/random.cpp
-    libcxx/src/regex.cpp
-    libcxx/src/shared_mutex.cpp
-    libcxx/src/stdexcept.cpp
-    libcxx/src/string.cpp
-    libcxx/src/strstream.cpp
-    libcxx/src/system_error.cpp
-    libcxx/src/thread.cpp
-    libcxx/src/typeinfo.cpp
-    libcxx/src/utility.cpp
-    libcxx/src/valarray.cpp
-    libcxx/src/variant.cpp
-    libcxx/src/vector.cpp
-    )
+target_compile_options(libcxx PRIVATE -std=c++14)
 
-target_compile_options(oelibcxx_cxx PRIVATE -std=c++14)
-target_compile_definitions(oelibcxx_cxx PRIVATE
-    -DLIBCXXRT
-    -D_LIBCPP_PROVIDES_DEFAULT_RUNE_TABLE
-    -D_LIBCPP_BUILDING_LIBRARY
-    -DOE_LIBC_SUPPRESS_DEPRECATIONS
-    )
+target_compile_definitions(libcxx
+  PRIVATE
+  -DLIBCXXRT
+  -D_LIBCPP_PROVIDES_DEFAULT_RUNE_TABLE
+  -D_LIBCPP_BUILDING_LIBRARY
+  -DOE_LIBC_SUPPRESS_DEPRECATIONS)
 
-# need the LIBCXX_CXX headers placed
-add_dependencies(oelibcxx_cxx oelibcxx_cxx_includes)
-# this project and its dependents require include/libcxx in the include path
-target_include_directories(oelibcxx_cxx PUBLIC
-    $<BUILD_INTERFACE:$<$<COMPILE_LANGUAGE:CXX>:${OE_INCDIR}/openenclave/libcxx>>
-    $<INSTALL_INTERFACE:$<$<COMPILE_LANGUAGE:CXX>:${CMAKE_INSTALL_INCLUDEDIR}/openenclave/3rdparty/libcxx>>
-    )
+target_link_libraries(libcxx PUBLIC oelibc)
 
-# this project (and is dependents) require the LIBC includes
-target_link_libraries(oelibcxx_cxx PUBLIC oelibc)
-# If we built an OBJECT library, we would want to do the following, though CMake does not allow
-#  using target_link_libraries on an OBJECT lib (yet?):
-#add_dependencies(oelibcxx_cxx oelibc)
-#target_include_directories(oelibcxx_cxx PUBLIC ${OE_INCDIR}/libc)
+target_include_directories(libcxx
+  PUBLIC
+  $<BUILD_INTERFACE:$<$<COMPILE_LANGUAGE:CXX>:${OE_INCDIR}/openenclave/libcxx>>
+  $<INSTALL_INTERFACE:$<$<COMPILE_LANGUAGE:CXX>:${CMAKE_INSTALL_INCLUDEDIR}/openenclave/3rdparty/libcxx>>
+  PRIVATE
+  libcxx/src
+  ../libcxxrt/libcxxrt/src)
 
-# when building this project, we also need a number of internal includes
-target_include_directories(oelibcxx_cxx PRIVATE libcxx/src)
-target_include_directories(oelibcxx_cxx PRIVATE ${CMAKE_SOURCE_DIR}/3rdparty/libcxxrt/libcxxrt/src)
+install(TARGETS libcxx EXPORT openenclave-targets
```

Comment:
It's supposed to allow us to generate a CMake package. However, we have an open issue noting that our generated CMake package is broken, so I'm going to audit and fix it in another step. For now, I tried to follow our existing logic as much as possible (because seemingly you have to export dependencies of exported targets, so I can't just delete this).

[Here is some information on how to make CMake packages](https://cmake.org/cmake/help/latest/manual/cmake-packages.7.html#creating-packages). The original devs mostly followed this, but it stopped being maintained, and so our export list is missing some necessary dependencies. See #468 and #887.

Reply:
Just to not, auditing this is going to be #887.